lib/govuk_index/payload_preparer.rb

```diff
@@ -22,7 +22,18 @@ def prepare_attachments(payload)
     end
 
     def prepare_parts(payload)
-      return payload unless payload["details"].fetch("parts", []).empty?
+      parts = payload["details"]["parts"]
+
+      if parts && parts.any?
+        updated_parts = parts.map do |part|
+          next part if part["link"]
+
+          link = "#{payload['base_path']}/#{part['slug']}"
+          part.merge("link" => link)
+        end
+
+        return merge_details(payload, "parts", updated_parts)
```

Comment:
I'm not sure I understand why this has to exit early?

I somewhat wonder if you need this code and line 48 could be:

```
"link" => "#{payload['base_path']}/#{part['slug']}",
```

?

---

I did wonder is it because parts["link"] may already be populated somehow? 

if so would it be a case that we need line 48 to be:

```
"link" => part.fetch("link", "#{payload['base_path']}/#{part['slug']}"),
```

?